CLTokenInputView/CLTokenInputView/CLTokenPillView.m

```diff
@@ -0,0 +1,152 @@
+//
+//  CLTokenPillView.m
+//  CLTokenInputView
+//
+//  Created by Doug Ross on 10/8/18.
+//  Copyright © 2018 Cluster Labs, Inc. All rights reserved.
+//
+
+#import "CLToken.h"
+#import "CLTokenPillView.h"
+
+static const CGFloat kPillSpacing = 8.0f;
+static const CGFloat kTitleHSpacing = 12.0f;
+static const CGFloat kTitleVSpacing = 5.0f;
+static const CGFloat kImageSpacing = 8.0f;
+static const CGFloat kImageWidth = 10.0f;
+static const CGFloat kImageHeight = 10.0f;
+
+@interface CLTokenPillView ()
+@property (strong, nonatomic) UILabel *title;
+@property (strong, nonatomic) UIView *pillView;
+@property (strong, nonatomic) UIImageView *dismissImage;
+@property (strong, nonatomic) UIView *dismissTapZone;
+@end
+
+@implementation CLTokenPillView
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithToken:(CLToken *)token font:(nullable UIFont *)font {
+    self = [super init];
+    if (self != nil) {
+        _token = token;
+        
+        _pillView = [UIView new];
+        _pillView.backgroundColor = UIColor.whiteColor;
+        _pillView.layer.borderColor = [UIColor colorWithRed:0.85f green:0.85f blue:0.85f alpha:1.0f].CGColor;
+        _pillView.layer.cornerRadius = 12.0f;
+
+        _pillView.layer.borderWidth = 1.0f;
+        [self addSubview:_pillView];
+        
+        _title = [[UILabel alloc] initWithFrame:CGRectMake(kTitleHSpacing, kTitleVSpacing, 0.0f, 0.0f)];
+        _title.font = font;
+        _title.text = token.displayText;
+        _title.textColor = [UIColor colorWithRed:0.25f green:0.25f blue:0.25f alpha:1.0f];
+        [_pillView addSubview:_title];
+        
+        _dismissImage = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"dismissX" inBundle:[NSBundle bundleForClass:[CLTokenPillView class]] compatibleWithTraitCollection:nil]];
+        _dismissImage.frame = CGRectMake(0.0f, 0.0f, kImageWidth, kImageHeight);
+        [_pillView addSubview:_dismissImage];
+        
+        _dismissTapZone = [UIView new];
+        _dismissTapZone.backgroundColor = UIColor.clearColor;
+        UITapGestureRecognizer *dismissTapRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissTapped:)];
+        [_dismissTapZone addGestureRecognizer:dismissTapRecognizer];
+        [_pillView addSubview:_dismissTapZone];
+        
```

Comment:
Instead of using tap gesture recognizers could this pill view be consisted of 2 buttons? The title could be within one of the buttons

Reply:
It was a pain getting things to line up using the button's various insets.  Since I had to use frame-based layout, it was easier to just specify the positioning directly.